### ISSUE TITLE
AttributeError: 'module' object has no attribute '_getframe'

### ISSUE THREAD

**Issue by zvacha** [IronLanguages/ironpython2#232]
I am getting this error when I try to install a library: Have used both of the following methods:
1. ipy -X:Frames -m pip install aosgenericdtelib
2. ipy setup.py install
I get the error in the subject. I am using Ironpython 2.7.7. I can install the library on a normal python 2.7 installation without any issues. Any help is very much appreciated

**Comment by slide** [comment]
You must use the first one, the second is not supported. Can you try adding -X:FullFrames instead of -X:Frames.

**Comment by slide** [comment]
Also, please paste the entire error message in the issue.

**Comment by zvacha** [comment]
Thank you for the help. One question:
1. Doesn't IronPython 2.7.7 not support installing from sources using setup.py?
Using FullFrames didn't help. Here is the error message
**
C:\temp\py2>ipy -X:FullFrames -m pip install aosgenericdtelib
Collecting aosgenericdtelib
C:\Program Files (x86)\IronPython 2.7\Lib\site-packages\pip\_vendor\requests\packages\urllib3\util\ssl_.py:1: SNIMissingWarning: An HTTPS request has been made, but the SNI (Subject Name Indication) extension to TLS is not available on this platform. This may cause the server to present an incorrect TLS certificate, which can cause validation failures. For more information, see https://urllib3.readthedocs.org/en/latest/security.html#snimissingwarning.
  from __future__ import absolute_import
  Using cached https://bln-vs-pypi.advaoptical.com/global/staged/+f/e39/2f93d52075b19/aosgenericdtelib-1.0.0.tar.gz
    Complete output from command python setup.py egg_info:
    Traceback (most recent call last):
      File "<string>", line 1, in <module>
      File "C:\Program Files (x86)\IronPython 2.7\Lib\site-packages\setuptools\__init__.py", line 11, in <module>
      File "C:\Program Files (x86)\IronPython 2.7\Lib\site-packages\setuptools\extern\__init__.py", line 1, in <module>
      File "C:\Program Files (x86)\IronPython 2.7\Lib\site-packages\pkg_resources\__init__.py", line 46, in <module>
      File "C:\Program Files (x86)\IronPython 2.7\Lib\site-packages\pkg_resources\extern\__init__.py", line 42, in load_module
      File "C:\Program Files (x86)\IronPython 2.7\Lib\site-packages\pkg_resources\_vendor\six.py", line 701, in <module>
      File "C:\Program Files (x86)\IronPython 2.7\Lib\site-packages\pkg_resources\_vendor\six.py", line 692, in exec_
    AttributeError: 'module' object has no attribute '_getframe'

    ----------------------------------------
Command "python setup.py egg_info" failed with error code 1 in c:\users\zvacha\appdata\local\temp\pip-build-bogzrc\aosgenericdtelib\
You are using pip version 8.1.1, however version 9.0.1 is available.
You should consider upgrading via the 'python -m pip install --upgrade pip' command.
**

**Comment by slozier** [comment]
There's an annoying bug where pip launches another instance of ipy and that instance doesn't run with frames enabled which is why you get the `_getframe` error. To get around this issue, you can replace the req_install.py file (in IronPython 2.7\Lib\site-packages\pip\req) with the attached patched version.

[req_install.zip](https://github.com/IronLanguages/ironpython2/files/1260345/req_install.zip)

**Comment by zvacha** [comment]
Thank you very much! This got around the _getframe error. I am getting another SystemError which I think is due to an issue with the library I am installing. But was wondering whether you can take a quick look at the following error message and comment. (I am fairly new with Python)

***
C:\Users\zvacha>ipy -X:FullFrames -m pip install aosgenericdtelib
Collecting aosgenericdtelib
C:\Program Files (x86)\IronPython 2.7\Lib\site-packages\pip\_vendor\requests\packages\urllib3\util\ssl_.py:1: SNIMissingWarning: An HTTPS request has been made, but the SNI (Subject Name Indication) extension to TLS is not available on this platform. This may cause the server to present an incorrect TLS certificate, which can cause validation failures. For more information, see https://urllib3.readthedocs.org/en/latest/security.html#snimissingwarning.
  from __future__ import absolute_import
  Using cached https://bln-vs-pypi.advaoptical.com/global/staged/+f/e39/2f93d52075b19/aosgenericdtelib-1.0.0.tar.gz
Collecting robotframework>=3.0 (from aosgenericdtelib)
  Using cached https://bln-vs-pypi.advaoptical.com/global/thirdparty/+f/201/8f233568227f9/robotframework-3.0.2.tar.gz
Collecting six (from aosgenericdtelib)
  Using cached https://bln-vs-pypi.advaoptical.com/root/pypi/+f/3ab/558cf5d4f7a72/six-1.10.0-py2.py3-none-any.whl
Collecting robotframework-aosdtedclilibrary (from aosgenericdtelib)
  Using cached https://bln-vs-pypi.advaoptical.com/global/staged/+f/678/4dc4fba272ca3/robotframework-aosdtedclilibrary-0.1.3.tar.gz
Collecting aos-cliclient (from aosgenericdtelib)
  Using cached https://bln-vs-pypi.advaoptical.com/global/staged/+f/3c5/66cecef272303/aos-cliclient-0.2.2.tar.gz
Collecting aos-dataparse (from aosgenericdtelib)
  Using cached https://bln-vs-pypi.advaoptical.com/global/staged/+f/8a7/e3c325111c805/aos-dataparse-0.1.7.tar.gz
Collecting robotframework-tools (from robotframework-aosdtedclilibrary->aosgenericdtelib)
  Using cached https://bln-vs-pypi.advaoptical.com/global/thirdparty/+f/2f1/8fcf2b757dd6f/robotframework-tools-1.0.tar.gz
Collecting robotframework-sshlibrary>=2.1.2 (from aos-cliclient->aosgenericdtelib)
  Using cached https://bln-vs-pypi.advaoptical.com/root/pypi/+f/878/6a6cf06382ad6/robotframework-sshlibrary-2.1.3.tar.gz
Collecting path.py>=8.0 (from robotframework-tools->robotframework-aosdtedclilibrary->aosgenericdtelib)
  Downloading https://bln-vs-pypi.advaoptical.com/root/pypi/+f/a02/7181918b884a7/path.py-10.4-py2.py3-none-any.whl
Collecting moretools>=0.1.8 (from robotframework-tools->robotframework-aosdtedclilibrary->aosgenericdtelib)
  Using cached https://bln-vs-pypi.advaoptical.com/root/pypi/+f/3a2/574fa069f8848/moretools-0.1.8.tar.gz
    Complete output from command python setup.py egg_info:
    Resolving setup requirement zetup >= 0.2.31:
    Traceback (most recent call last):
      File "<string>", line 1, in <module>
      File "c:\users\zvacha\appdata\local\temp\pip-build-fphvpv\moretools\setup.py", line 65, in <module>
      File "C:\Program Files (x86)\IronPython 2.7\Lib\site-packages\setuptools\command\easy_install.py", line 1078, in build_and_install
      File "c:\users\zvacha\appdata\local\temp\pip-build-fphvpv\moretools\setup.py", line 56, in resolve
      File "C:\Program Files (x86)\IronPython 2.7\Lib\site-packages\setuptools\dist.py", line 380, in fetch_build_egg
      File "C:\Program Files (x86)\IronPython 2.7\Lib\site-packages\setuptools\command\easy_install.py", line 640, in easy_install
      File "C:\Program Files (x86)\IronPython 2.7\Lib\site-packages\setuptools\command\easy_install.py", line 670, in install_item
      File "C:\Program Files (x86)\IronPython 2.7\Lib\site-packages\setuptools\command\easy_install.py", line 850, in install_eggs
      File "C:\Program Files (x86)\IronPython 2.7\Lib\site-packages\setuptools\command\easy_install.py", line 1064, in run_setup
      File "C:\Program Files (x86)\IronPython 2.7\Lib\site-packages\setuptools\sandbox.py", line 243, in run_setup
      File "C:\Program Files (x86)\IronPython 2.7\Lib\contextlib.py", line 35, in __exit__
      File "C:\Program Files (x86)\IronPython 2.7\Lib\contextlib.py", line 35, in __exit__
      File "C:\Program Files (x86)\IronPython 2.7\Lib\site-packages\setuptools\sandbox.py", line 178, in save_pkg_resources_state
    SystemError: Parent module 'zetup' not loaded, cannot perform relative import

    ----------------------------------------
Command "python setup.py egg_info" failed with error code 1 in c:\users\zvacha\appdata\local\temp\pip-build-fphvpv\moretools\
You are using pip version 8.1.1, however version 9.0.1 is available.
You should consider upgrading via the 'python -m pip install --upgrade pip' command.
****

**Comment by zvacha** [comment]
Thank you!

**Comment by RobAtSNC** [comment]
Even with req_install.py from @slozier, I was getting the missing `_getframe` error when using the `-e` option.
```bash
ipy -X:Frames -m pip install pip install -e path-to-my-package
```
I fixed it with further modification to req_install.py.
```diff
--- req_install-old.py  2017-09-12 13:01:17.204423600 -0600
+++ req_install.py      2017-09-12 13:01:01.163675000 -0600
@@ -975,12 +975,15 @@

         with indent_log():
             # FIXME: should we do --install-headers here too?
-            call_subprocess(
-                [
+            cmd = [
                     sys.executable,
                     '-c',
                     SETUPTOOLS_SHIM % self.setup_py
-                ] +
+                  ]
+            if sys.platform == 'cli':
+                cmd.insert(1, '-X:Frames')
+            call_subprocess(
+                cmd +
                 list(global_options) +
                 ['develop', '--no-deps'] +
                 list(install_options),
```
[req_install.zip](https://github.com/IronLanguages/ironpython2/files/1297131/req_install.zip)

**Comment by slide** [comment]
The missing _getframe issue should be resolved with 2.7.8 because we enable frames by default.

**Comment by slide** [comment]
2.7.8 enabled frames by default, so this error should not occur.
